internal/services/mssql/mssql_job_schedule_resource.go

```diff
@@ -0,0 +1,319 @@
+	Interval  string `tfschema:"interval"`
+	JobID     string `tfschema:"job_id"`
+	StartTime string `tfschema:"start_time"`
+	Type      string `tfschema:"type"`
+}
+
+var (
+	_ sdk.ResourceWithUpdate        = MsSqlJobScheduleResource{}
+	_ sdk.ResourceWithCustomizeDiff = MsSqlJobScheduleResource{}
+)
+
+func (MsSqlJobScheduleResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			// Note: O+C the API sets this field to `false` if the schedule type is `Once` and the job has finished
+			Computed: true,
+		},
+		"end_time": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			// Note: O+C API sets and returns a default value if omitted
+			Computed:         true,
+			DiffSuppressFunc: suppress.RFC3339MinuteTime,
+			ValidateFunc:     validation.IsRFC3339Time,
+		},
+		"interval": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validate.ISO8601Duration,
+		},
+		"job_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: jobs.ValidateJobID,
+		},
+		"start_time": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			// Note: O+C API sets and returns a default value if omitted
+			Computed:         true,
+			DiffSuppressFunc: suppress.RFC3339MinuteTime,
+			ValidateFunc:     validation.IsRFC3339Time,
+		},
+		"type": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringInSlice(jobs.PossibleValuesForJobScheduleType(), false),
+		},
+	}
+}
+
+func (MsSqlJobScheduleResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (MsSqlJobScheduleResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 10 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var config MsSqlJobScheduleResourceModel
+			if err := metadata.DecodeDiff(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			if config.Type == string(jobs.JobScheduleTypeRecurring) && config.Interval == "" {
+				return fmt.Errorf("`interval` must be set when `type` is `Recurring`")
+			}
+
+			return nil
+		},
+	}
+}
+
+func (MsSqlJobScheduleResource) ModelObject() interface{} {
+	return &MsSqlJobScheduleResourceModel{}
+}
+
+func (MsSqlJobScheduleResource) ResourceType() string {
+	return "azurerm_mssql_job_schedule"
+}
+
+func (r MsSqlJobScheduleResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MSSQL.JobsClient
+
+			var config MsSqlJobScheduleResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			jobId, err := jobs.ParseJobID(config.JobID)
+			if err != nil {
+				return err
+			}
+
```

Comment:
Since this a virtual resource (by virtual I mean it's not a resource in Azure in the sense that it has it's own ID and endpoints, we've just modeled it as a resource in the provider) that's modifying a parent resource, we should add locks on the parent resource. This will prevent Terraform from trying to update the resource in two different places which can lead to problems.
```suggestion
	locks.ByID(jobId.ID())
	defer locks.UnlockByID(jobId.ID())
```